internal/services/fetcher.go

```diff
@@ -0,0 +1,98 @@
+package services
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/shopspring/decimal"
+	"github.com/shulganew/gophermart/internal/api/client"
+	"github.com/shulganew/gophermart/internal/config"
+	"github.com/shulganew/gophermart/internal/model"
+	"go.uber.org/zap"
+)
+
+type Fetcher struct {
+	stor FetcherUpdater
+	conf *config.Config
+
+	// map[order number]Order
+	orders map[string]model.Order
+	mu     sync.Mutex
+}
+
+type FetcherUpdater interface {
+	LoadPocessing(ctx context.Context) ([]model.Order, error)
+	UpdateStatus(ctx context.Context, order string, status model.Status) (err error)
+	SetAccrual(ctx context.Context, order string, accrual decimal.Decimal) (err error)
+	AddBonuses(ctx context.Context, userID uuid.UUID, amount decimal.Decimal) (err error)
+}
+
+func NewFetcher(stor FetcherUpdater, conf *config.Config) *Fetcher {
```

Comment:
```suggestion
func NewAccrualService(
  orderRepo OrderRepository, 
  accrualClient AccrualClient, 
  conf *config.Config
) *Fetcher {
```
тут содержится логика обновления заказов с помощью сервиса accrual - давай так и назовем. Фетчер очень абстрактно - фетчить можно что угодно. Файлик можно назвать accrual.go

Reply:
Готово